sklearn/svm/tests/test_sparse.py

```diff
@@ -111,10 +111,19 @@ def test_unsorted_indices():
     # make sure dense and sparse SVM give the same result
     assert_array_almost_equal(coef_dense, coef_sorted.toarray())
 
-    X_sparse_unsorted = X_sparse[np.arange(X.shape[0])]
-    X_test_unsorted = X_test[np.arange(X_test.shape[0])]
+    # scramble the indices
+    def scramble_indices(X):
+        new_data = []
+        new_indices = []
+        for i in range(1, len(X.indptr)):
+            new_data.extend(X.data[X.indptr[i - 1]: X.indptr[i]][::-1])
```

Comment:
This would be more readable if you pulled out the shared `indptr` parts:
```python
row_slice = slice(*X.indptr[i-1:i+1])
new_data.extend(X.data[row_slice][::-1])
...
```